Add classifiers to setup.py and drop Python 2.6 support

cdeil opened the pull request:
This PR adds classifiers to `setup.py` ... the main point is to declare which Python versions we support in the standard way packaging tools understand.
E.g. currently [photutils on PyPI](https://pypi.python.org/pypi/photutils) has no such info, but [astropy on PyPI](https://pypi.python.org/pypi/astropy/0.4.2) has it.

I also suggest we remove support for Python 2.6 but @astrofrog, @larrybradley if you think there are still users that need to run on Python 2.6 I'll put it that travis-ci build back in.

bsipocz commented:
I think we can also remove the build No. 6 (``ASTROPY_VERSION=development SETUP_CMD='test'``), as it is included in No4 (``ASTROPY_VERSION=development SETUP_CMD='test --coverage'``)

astrofrog commented:
@cdeil - since Astropy still supports 2.6, I don't see the harm in continuing to test 2.6 here. I think we should consider dropping 2.6 in future if not dropping it would mean actual convoluted code, but until then, what do we lose by testing this on Travis?

cdeil commented:
@bsipocz - I've removed the redundant travis-ci build you mentioned.

@astrofrog - I've added back Python 2.6 on travis-ci and listed it in the supported Python versions in `setup.py`.

I've had (admittedly very few) real issues only surface on Python 2.6 in Astropy and Gammapy, and it takes some build time on travis-ci ... that was my motivation for suggesting to drop support for it now.
But OK ... if you think there's users that need to run it on 2.6 (i.e. there's any value in supporting it for `photutils 0.2`), let's keep Python 2.6 support.

larrybradley commented:
I agree it's probably good to stay in sync with `astropy`'s requirements (especially if `photutils` eventually goes into the `astropy` core).

cdeil commented:
Ready to merge or do you want some classifiers added or removed?

larrybradley commented:
Looks good to me, :+1:.

astrofrog commented:
:+1:
